core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -416,6 +416,9 @@ public void updateRoutingEntry(ShardRouting newRouting) throws IOException {
                 // active primaries.
                 throw new IndexShardRelocatedException(shardId(), "Shard is marked as relocated, cannot safely move to state " + newRouting.state());
             }
+            assert newRouting.active() == false || state == IndexShardState.STARTED || state == IndexShardState.RELOCATED ||
+                state == IndexShardState.CLOSED :
+                "shard state is " + state + ", but routing is active " + newRouting;
```

Comment:
OK. got it. I got confusing. I would prefer a message like "routing is active, but local shard state isn't.  routing [] , local state [ ]". If you prefer yours I'm got with keeping as is.